### PR TITLE
Refactored

### PR DESCRIPTION
**Pull request by arun-maitaitech**

### CODE REVIEW

File: app/globalTeardown.ts [hunk 1]
```diff
@@ -4,8 +4,11 @@ import { consoleDebug } from './utils';
 
 
 export default async () => {
-  const mainDataSource = getMainDataSource();
-  consoleDebug(`globalTeardown - wasEverInitialized: ${mainDataSource.wasEverInitialized()}`);
-  await mainDataSource.closeAndDelete_templateDb();
+  let  mainDataSource = getMainDataSource();
+  mainDataSource.createOrConnectToExistingTemplateDB();
```
Comment (Asaf-S) [inline app/globalTeardown.ts:8]: That's not the time to create the template DB for the first time.
Can't we just connect to it if it exists?

File: app/index.ts [hunk 2]
```diff
@@ -68,22 +68,24 @@ const unwrap = (testName: string, userFn: (dbData: { dbNameForThisTest: string;
 };
 
 // Initialize the DB
-void getMainDataSource().createTemplateDB();
+void getMainDataSource().createOrConnectToExistingTemplateDB();
```
Comment (Asaf-S) [inline app/index.ts:71]: Why should we connect to an existing DB here?
This is the place to create a new template DB, no?

**Review by Asaf-S — APPROVED** [review]

**Comment by Asaf-S** [comment]
Please also add the required file to the `demo` project

File: app/globalTeardown.ts (hunk 1 above)
Comment (arun-maitaitech) [reply app/globalTeardown.ts:8]: Corrected : https://github.com/Asaf-S/jest-test-clean-db/pull/3

**Comment by arun-maitaitech** [comment]
> Please also add the required file to the `demo` project. Respond "Done" to this comment, as it cannot be resolved.

Done.

**Pull request merged**